crates/turborepo-lib/src/run/summary/mod.rs

```diff
@@ -48,67 +69,122 @@ enum RunType {
     DryJson,
 }
 
-fn get_user(env_vars: &EnvironmentVariableMap) -> Option<String> {
-    if turborepo_ci::is_ci() {
-        return Vendor::get_info()
-            .and_then(|vendor| vendor.username_env_var)
-            .and_then(|username_env_var| env_vars.get(username_env_var).cloned());
-    }
-
-    None
+// Can't reuse `cli::EnvMode` because the serialization
+// is different (lowercase vs uppercase)
+#[derive(Clone, Copy, Debug, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum EnvMode {
+    Infer,
+    Loose,
+    Strict,
 }
 
-// Wrapper around the serializable RunSummaryInner, with some extra information
-// about the Run and references to other things that we need.
-#[derive(Debug)]
-pub struct RunSummary<'a> {
-    inner: RunSummaryInner<'a>,
-    repo_root: &'a AbsoluteSystemPath,
-    single_package: bool,
-    should_save: bool,
-    run_type: RunType,
-    synthesized_command: String,
+impl From<cli::EnvMode> for EnvMode {
+    fn from(env_mode: cli::EnvMode) -> Self {
+        match env_mode {
+            cli::EnvMode::Infer => EnvMode::Infer,
+            cli::EnvMode::Loose => EnvMode::Loose,
+            cli::EnvMode::Strict => EnvMode::Strict,
+        }
+    }
 }
 
 #[derive(Debug, Serialize)]
 #[serde(rename_all = "camelCase")]
-pub struct RunSummaryInner<'a> {
+pub struct RunSummary<'a> {
     id: Ksuid,
     version: String,
     turbo_version: String,
     monorepo: bool,
+    #[serde(rename = "globalCacheInputs")]
     global_hash_summary: GlobalHashSummary<'a>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    execution: Option<ExecutionSummary<'a>>,
     packages: HashSet<WorkspaceName>,
     env_mode: EnvMode,
     framework_inference: bool,
-    execution_summary: Option<ExecutionSummary<'a>>,
     tasks: Vec<TaskSummary<'a>>,
-    user: Option<String>,
+    user: String,
+    pub scm: SCMState,
```

Comment:
I'm not seeing us reference this field from outside this module
```suggestion
    scm: SCMState,
```